apollo-router/src/spec/query.rs

```diff
@@ -837,10 +820,6 @@ impl Query {
                         );
                         path.pop();
                         res?
-                    } else if name.as_str() == TYPENAME {
-                        if !output.contains_key(field_name_str) {
-                            output.insert(field_name.clone(), Value::String(root_type_name.into()));
-                        }
```

Comment:
Move it in front as the first option since the subgraph can have different names of root type, but the root type is always object type, so we ignore value returned from the subgraph by moving this code above.